artichoke-backend/src/convert/string.rs

```diff
@@ -128,6 +129,7 @@ mod tests {
         }
 
         #[allow(clippy::needless_pass_by_value)]
+        #[cfg(feature = "core-regexp")]
```

Comment:
Note to self: This feature dependency is because `String#[]` is implemented in pure Ruby and depends on `Regexp`.

https://github.com/artichoke/artichoke/blob/44b60ed9a8f3922cfac6c8e07ecc343e3091bc3e/artichoke-backend/src/extn/core/string/string.rb#L184-L197